# TASKS


This notebook contains the tasks for this subject.
Each task contains a different problem to solve, which will include a brief description of the problem, research content, the result of the exercise and personal conclusions when performing it.

***

## Task nº1 - Lady Tasting Tea

### Task statement

Suppose we alter the Lady Tasting Tea experiment to involve twelve cups of tea. Six have the milk in first and the other six having tea in first. A person claims they have the special power of being able to tell whether the tea or the milk went into a cup first upon tasting it. You agree to accept their claim if they can tell which of the six cups in your experiment had the milk in first.

Calculate, using Python, the probability that they select the correct six cups. Here you should assume that they have no special powers in figuring it out, that they are just guessing. Remember to show and justify your workings in code and MarkDown cells.

Suppose, now, you are willing to accept one error. Once they select the six cups they think had the milk in first, you will give them the benefit of the doubt should they have selected at least five of the correct cups. Calculate the probability, assuming they have no special powers, that the person makes at most one error.

Would you accept two errors? Explain.

### Information about the task, explanation and approach used

This experiment presented as task number one is a variation on one of the many statistical problems contained in the book called "The Design of Experiments" which was written in 1935 by the English statistician **Ronald Fisher**, on the design of experiments and is considered a fundamental work in experimental design and in modern statistical science.

In this experiment, Fisher introduced the concept of the null hypothesis into the framework of hypothesis testing as part of his approach to the analysis of experiments.

This procedure can be applied in many contexts, and is a fundamental basis in statistics for making data-based decisions.

To understand the application of this theory in the experiment, I am going to indicate a brief description of the null hypothesis and its application to this case.

The null hypothesis (H₀) and the alternative hypothesis (H₁) are basic concepts used in inferential statistics for decision making.

We try to find evidence to support or refute assumptions or statements.

\begin{align*}
H_0 &: \theta = \theta_0 \\
H_1 &: \theta \neq \theta_0
\end{align*}

**Null hypothesis (H₀)**: It is the initial statement that is assumed to be true until proven otherwise.  
Applied to our experiment, it may consist of the assumption that the person drinking tea does not have any sixth sense, that is, that person guesses how that tea was prepared.

The **alternative hypothesis((H₁))** is the statement contrary to the null hypothesis, that is, it is what the experiment is trying to prove.  
In our example, the person has a gift that goes beyond good luck, he is quite a fortune teller.

To solve this problem, we can choose between the use of permutations or combinations, however I will opt for the use of combinations, since the problem establishes that the order of the cups is not important and the combinations are preferable because it is only interesting the selection of cups, not the order.  

The formula used for the combinations is the following:  

$$\binom{n}{k} = \frac{n!}{k!(n-k)!}$$ 

where:  

"n" is the number of elements, and i this particular example is 12  
"k" is the number of selected elements,6 cups that have the milk first  
$\binom{n}{k}$ is the number of ways to choose "k" elements on a set of "n" elements

To evaluate the evidence against the null hypothesis, a concept called **p - value** is used.

If the p value meets the condition of being less than an arbitrarily imposed significance level, this is considered a statistically significant result and, therefore, allows us to reject the null hypothesis.  

The significance level "α" is a threshold that is chosen in advance before performing the experiment and is commonly set at 0.05 (5%).  
This means that a 5% probability of incorrectly rejecting the null hypothesis is accepted.

For the code part,I am going to use several functionalities of the python numpy library and the matplotlib library to more directly visualize the results of my analysis.

### Problem statement

1) For this experiment we have 12 cups of tea of  ​​which 6 have milk first and the remaining ones have tea first.
2) A person claims they have the special power of being able to tell whether the tea or the milk went into a cup first upon tasting it.
3) Calculate the probability that the person guesses all the cups by chance. This is our null hypothesis(the person does not have any special skill).
4) Calculate the probability, assuming they have no special powers, that the person makes at most one error(correctly select 5 cups out of 6).
5) The order when guessing the cups is not important.

In [1]:
# Library Imports
import numpy as np
import matplotlib.pyplot as plt

Once I import the libraries variables are set and with them I calculate the combinations by calculating the probability that the person choose the correct 6 cups by chance

In [2]:
total_cups = 12
correct_cups = 6  # 6 cups that have the milk first

In [3]:
# Creation of factorial products for the calculation of combinations.
def combinations(n, k):
    return np.prod(np.arange(n - k + 1, n + 1)) // np.prod(np.arange(1, k + 1))

Using "n" and "k" instead of the variables we created at the beginning, I can modify the number of cups if I want.  
Now I calculate the total of possible combinations and the probability that the person has to choose the correct 6 in a row.

In [4]:
total_combinations = combinations(total_cups, correct_cups)
prob_correct = 1 / total_combinations
print(f"The number of possible combinations is : {total_combinations}")
print(f"The probability that they select the correct six cups: {prob_correct:.5f}")

The number of possible combinations is : 924
The probability that they select the correct six cups: 0.00108


That is, it gives us a probability of 0.108225% that it is correct when choosing the 6 cups among all the available combinations.  


To evaluate our hypothesis I use the p value mentioned above and compare it with the obtained result.  
  

If p < α I reject the null hypothesis  

As "p" in the example is 0,00108 and "α" is 0,05 that means 0,00108 < 0,05 **the null hypothesis would be rejected.**  

I reject the null hypothesis in favor of the alternative hypothesis, indicating that the person who tastes the teas has a special ability.


For the second part of the problem, it is stated that the person makes at most one error and that he does not have special skills, that is, that he chooses at least 5 cups correctly.  
Since it is one cup less, the probability will be greater, but I will have to calculate it to know if this time I reject the null hypothesis again.

In [5]:
# Give the number of ways in which 5 correct cups and 1 incorrect cup can be chosen
comb_5_correct = combinations(correct_cups, 5) * combinations(total_cups - correct_cups, 1) 
prob_5_correct = (comb_5_correct + 1) / total_combinations

The problem also asks if I could accept 2 errors, so I am going to calculate this new situation and then compare it with the previous one.

In [6]:
# Give the number of ways in which 4 correct cups and 2 incorrect cup can be chosen
comb_4_correct = combinations(correct_cups, 4) * combinations(total_cups - correct_cups, 2)
prob_4_correct = comb_4_correct / total_combinations

Now with the calculations done, tt's time to interpret the results.  

### Analysis results
A compilation of the results to establish which hypotheses are rejected or not.

In [7]:
print(f"The probability of selecting the correct six cups: {prob_correct:.5f}")
print(f"The probability of selecting 5 cups correctly with 1 error: {prob_5_correct:.5f}")
print(f"The probability of selecting 4 cups correctly with 2 errors : {prob_4_correct:.5f}")

The probability of selecting the correct six cups: 0.00108
The probability of selecting 5 cups correctly with 1 error: 0.04004
The probability of selecting 4 cups correctly with 2 errors : 0.24351


For the first option, I rejected the null hypothesis.  
For the second option, as 0.04004 < 0.05 I also reject the null hypothesis.  
For the last option,  0.24351 > 0.05 , so in this case the null hypothesis is not rejected.  
This means that it is possible for a participant who has no special skills to guess 4 cups correctly by chance.

For the first and second cases, the probability is very low, suggesting that the person does not have any special abilities.
However, for the third option, it is possible but not totally certain that by accepting 2 errors, the person has special abilities.  
I could consider accepting 2 errors with the probability of 24.351% but the fact that one has special powers is not conclusive.